examples/applications/plot_stock_market.py

```diff
@@ -158,50 +110,94 @@
 close_prices = np.vstack([q["close"] for q in quotes])
 open_prices = np.vstack([q["open"] for q in quotes])
 
-# The daily variations of the quotes are what carry most information
+# The daily variations of the quotes are what carry the most information
 variation = close_prices - open_prices
 
+# %%
+# Learning a graph structure
+# --------------------------
+#
+# We use sparse inverse covariance estimation to find which quotes are
+# correlated conditionally on the others. Specifically, sparse inverse
+# covariance gives us a graph, that is a list of connection. For each
+# symbol, the symbols that it is connected too are those useful to explain
+# its fluctuations.
 
-# #############################################################################
-# Learn a graphical structure from the correlations
 edge_model = covariance.GraphicalLassoCV()
 
 # standardize the time series: using correlations rather than covariance
-# is more efficient for structure recovery
+# former is more efficient for structure recovery
 X = variation.copy().T
 X /= X.std(axis=0)
 edge_model.fit(X)
```

Comment:
Yeah, I am still getting the warning. I looked into it and it is coming from the NumPy. This warning is generated if we pass a non desirable value in the `x = asanyarray(arr-arrmean)` [NumPy code here line 232](https://github.com/numpy/numpy/blob/main/numpy/core/_methods.py). You should be getting that as well in the warning. I checked the values and somehow the `arr` and `arrmean` have many `Nan` and `-inf` values in it. That could be the reason for this warning. I am looking into it why this is happening. Any suggestions there might be helpful.

Reply:
Oh I forgot to mention what was the solution here. Indeed, we try a number of `alphas` values and some of those alphas are not suited and lead to `nan`.

The solution is then to pass an `alphas=np.logspace(-2, 0, num=10)` for instance. With the fix merged in https://github.com/scikit-learn/scikit-learn/pull/22493 you should not have any problem now. You will only need to merge `main` into your branch to be sure that passing the NumPy array will not raise an error.

Reply:
Thanks for the update!

Reply:
I am facing an error while running this example. It is not able to import `PairwiseDistancesArgKmin`.
```
Traceback (most recent call last):
  File "plot_stock_market.py", line 119, in <module>
    from sklearn import covariance
  File "d:\aditya's stuff\open source projects\scikitlearn_contribution\scikit-learn\sklearn\covariance\__init__.py", line 9, in <module>
    from ._empirical_covariance import (
  File "d:\aditya's stuff\open source projects\scikitlearn_contribution\scikit-learn\sklearn\covariance\_empirical_covariance.py", line 21, in <module>
    from ..metrics.pairwise import pairwise_distances
  File "d:\aditya's stuff\open source projects\scikitlearn_contribution\scikit-learn\sklearn\metrics\__init__.py", line 41, in <module>
    from . import cluster
  File "d:\aditya's stuff\open source projects\scikitlearn_contribution\scikit-learn\sklearn\metrics\cluster\__init__.py", line 22, in <module>
    from ._unsupervised import silhouette_samples
  File "d:\aditya's stuff\open source projects\scikitlearn_contribution\scikit-learn\sklearn\metrics\cluster\_unsupervised.py", line 16, in <module>
    from ..pairwise import pairwise_distances_chunked
  File "d:\aditya's stuff\open source projects\scikitlearn_contribution\scikit-learn\sklearn\metrics\pairwise.py", line 35, in <module>
    from ._pairwise_distances_reduction import PairwiseDistancesArgKmin
ModuleNotFoundError: No module named 'sklearn.metrics._pairwise_distances_reduction'
```

I think we have to Cythonize the _pairwise_distances_reduction.pyx file to get the .c and .pyd file extensions which we can import. Because without it I am getting an import error here.

Reply:
I am not able to build the `.pyd` file for this. Should I open a new issue for this?